From eda8b578bd7e460242b1f458faf89ed52b4538d5 Mon Sep 17 00:00:00 2001
From: Wei-Luan Wang <weiluanwang@google.com>
Date: Thu, 26 Jun 2025 03:06:46 +0000
Subject: [PATCH] disable boringssl used in
 base::NonAllocatingRandomBitGenerator

Introduced in crrev.com/1057955: libchrome does not support boringssl.

Change-Id: I71b29173282129b5b61a93971ef5dd24ee6b60a9
patch-name: long-term-2000-disable-boringssl-used-in-base-NonAllocatingRandomBitGenerator.patch
---
 base/rand_util.h | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/base/rand_util.h b/base/rand_util.h
index 518ae6addb..377f6c7e6b 100644
--- a/base/rand_util.h
+++ b/base/rand_util.h
@@ -23,7 +23,6 @@
 #include "base/numerics/safe_conversions.h"
 #include "base/time/time.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/rand.h"
 
 namespace memory_simulator {
 class MemoryHolder;
@@ -186,22 +185,6 @@ class RandomBitGenerator {
   ~RandomBitGenerator() = default;
 };
 
-class NonAllocatingRandomBitGenerator {
- public:
-  using result_type = uint64_t;
-  static constexpr result_type min() { return 0; }
-  static constexpr result_type max() { return UINT64_MAX; }
-  result_type operator()() const {
-    uint64_t result;
-    RAND_get_system_entropy_for_custom_prng(reinterpret_cast<uint8_t*>(&result),
-                                            sizeof(result));
-    return result;
-  }
-
-  NonAllocatingRandomBitGenerator() = default;
-  ~NonAllocatingRandomBitGenerator() = default;
-};
-
 // Shuffles [first, last) randomly. Thread-safe.
 template <typename Itr>
 void RandomShuffle(Itr first, Itr last) {
-- 
2.50.0.727.gbf7dc18ff4-goog

